lang/cs/Org.Apache.REEF.Network/Naming/Events/NamingEvent.cs

```diff
@@ -25,7 +25,7 @@ namespace Org.Apache.REEF.Network.Naming.Events
     /// Event representing a lookup, registering, or unregistering of 
     /// an identifier with the Name Service.
     /// </summary>
-    internal class NamingEvent
+    internal abstract class NamingEvent
```

Comment:
That would mean that `public ILink<NamingEvent> Link { get; set; }` will need to be duplicated everywhere in all its subclasses. I think `abstract` may be more suitable here.